src/main/java/org/folio/circulation/resources/RequestByInstanceIdResource.java

```diff
@@ -0,0 +1,322 @@
+
+    availableItems.thenCompose(r -> r.after( collectionResult -> findItemsWithMatchingServicePointId(
+                                                                  pickupServicePointId, collectionResult, locationRepository)))
+      .thenApply( r -> r.next( itemsFound -> instanceToItemRequests(requestByInstanceIdRequest, itemsFound)))
+      .thenCompose( r -> r.after( requests -> placeRequests(requests, clients)))
+      .thenApply(r -> r.map(RequestAndRelatedRecords::getRequest))
+      .thenApply(r -> r.map(new RequestRepresentation()::extendedRepresentation))
+      .thenApply(CreatedJsonResponseResult::from)
+      .thenAccept(result -> result.writeTo(routingContext.response()))
+      .exceptionally( err -> {
+          log.error("Error processing title-level request", err);
+          return null;});
+  }
+
+
+  private CompletableFuture<Result<RequestAndRelatedRecords>> placeRequests(List<JsonObject> itemRequestRepresentations, Clients clients) {
+
+    final RequestNoticeSender requestNoticeSender = RequestNoticeSender.using(clients);
+
+    final CreateRequestService createRequestService = new CreateRequestService(
+      RequestRepository.using(clients),
+      new UpdateItem(clients),
+      new UpdateLoanActionHistory(clients),
+      new UpdateLoan(clients, loanRepository, new LoanPolicyRepository(clients)),
+      new RequestPolicyRepository(clients),
+      loanRepository, requestNoticeSender);
+
+    return placeRequest(itemRequestRepresentations, 0, createRequestService);
+  }
+
+  private CompletableFuture<Result<RequestAndRelatedRecords>> placeRequest(List<JsonObject> itemRequests, int startIndex,
+                                                              CreateRequestService createRequestService) {
+    if (startIndex >= itemRequests.size()) {
+      return CompletableFuture.completedFuture(failed(new ServerErrorFailure(
+        "Failed to place a request for the title")));
+    }
+
+    JsonObject currentItemRequest = itemRequests.get(startIndex);
+
+    final RequestFromRepresentationService requestFromRepresentationService =
+      new RequestFromRepresentationService(
+        new ItemRepository(clients, true, false, false),
+        RequestQueueRepository.using(clients),
+        userRepository,
+        loanRepository,
+        new ServicePointRepository(clients),
+        createProxyRelationshipValidator(currentItemRequest, clients),
+        new ServicePointPickupLocationValidator()
+      );
+
+    return requestFromRepresentationService.getRequestFrom(currentItemRequest)
+      .thenCompose(r -> r.after(createRequestService::createRequest))
+      .thenCompose(r -> {
+          if (r.succeeded()) {
+            return CompletableFuture.completedFuture(r);
+          } else {
+            log.debug("Failed to create request for " + currentItemRequest.getString("id"));
+            return placeRequest(itemRequests, startIndex +1, createRequestService);
```

Comment:
My concern with this approach is that request creation might fail for a variety of reasons, and not because the request is invalid. 

In those circumstances, the system will move on to the next item. It might partially create multiple requests.

I likely didn't manage to raise this early enough in the process, in response to your thorough early design comments, and it is likely too late to change it. As it would likely require significant refactoring of the existing code to make this more amenable to doing it without potential side effects.

Reply:
I'm hoping that we will get a chance to implement the feature for real (not this pseudo implementation) and we will get to do what is right.  For this simple implementation, it might not be worth it to do significant refactoring. However, if you know of a case that illustrates this example, I'd love to see how this code reacts to it.

Reply:
Thinking of an example, is relative easy. Recreating it in the system less so.

If the update of the item status or loan fails, then I imagine this code would move on to the next item. That could result in each item getting a new request.

I agree that it likely requires a lot of refactoring. The validation of the request would need separating from the creation. Then only when the validation succeeded, actually create the request.

Reply:
(I reckon there is significant complexity around a mixed instance and item queue)